Infrastructure/Services/ParcelService.cs

```diff
@@ -117,5 +118,62 @@ public async Task<Parcel> Update(string trackingId, Parcel updatedParcel)
                 throw;
             }
         }
+        public async Task<Parcel> UpdateParcelStatus(string trackingId, Status status)
```

Comment:
I'd rename trackingId to trackingNumber for consistency everywhere. Also I think that methods that update parcels by id would make sense - tracking number is more for display purpose to the user ( at least that is how I imagine it), of course these can be easily added later if need be